x-pack/test/alerting_api_integration/security_and_spaces/group1/tests/alerting/backfill/task_runner.ts

```diff
@@ -0,0 +1,769 @@
+            immutable: false,
+            license: '',
+            outputIndex: '',
+            meta: {
+              from: '1m',
+              kibana_siem_app_url: 'https://localhost:5601/app/security',
+            },
+            maxSignals: 100,
+            riskScore: 21,
+            riskScoreMapping: [],
+            severity: 'low',
+            severityMapping: [],
+            threat: [],
+            to: 'now',
+            references: [],
+            version: 1,
+            exceptionsList: [],
+            relatedIntegrations: [],
+            requiredFields: [],
+            setup: '',
+            type: 'query',
+            language: 'kuery',
+            index: [ES_TEST_INDEX_NAME],
+            query: `source:${DOCUMENT_SOURCE}`,
+            filters: [],
+          },
+        })
+        .expect(200);
+      const ruleId = response1.body.id;
+      objectRemover.add(spaceId, ruleId, 'rule', 'alerting');
+
+      // Schedule backfill for this rule
+      // schedule backfill for both rules as current user
+      const response2 = await supertestWithoutAuth
+        .post(`${getUrlPrefix(spaceId)}/internal/alerting/rules/backfill/_schedule`)
+        .set('kbn-xsrf', 'foo')
+        .auth(SuperuserAtSpace1.user.username, SuperuserAtSpace1.user.password)
+        .send([
+          {
+            rule_id: ruleId,
+            start: '2023-10-19T12:00:00.000Z',
+            end: '2023-10-21T12:00:00.000Z',
+          },
+        ])
+        .expect(200);
+
+      const scheduleResult = response2.body;
+
+      expect(scheduleResult.length).to.eql(1);
+      expect(scheduleResult[0].schedule.length).to.eql(4);
+      expect(scheduleResult[0].schedule).to.eql([
+        {
+          interval: '12h',
+          run_at: '2023-10-20T00:00:00.000Z',
+          status: 'pending',
+        },
+        {
+          interval: '12h',
+          run_at: '2023-10-20T12:00:00.000Z',
+          status: 'pending',
+        },
+        {
+          interval: '12h',
+          run_at: '2023-10-21T00:00:00.000Z',
+          status: 'pending',
+        },
+        {
+          interval: '12h',
+          run_at: '2023-10-21T12:00:00.000Z',
+          status: 'pending',
+        },
+      ]);
+
+      const backfillId = scheduleResult[0].id;
+
+      // check that the task was scheduled correctly
+      const taskRecord = await getScheduledTask(backfillId);
+      expect(taskRecord.type).to.eql('task');
+      expect(taskRecord.task.taskType).to.eql('ad_hoc_run-backfill');
+      expect(taskRecord.task.timeoutOverride).to.eql('5m');
+      expect(taskRecord.task.enabled).to.eql(true);
+      expect(JSON.parse(taskRecord.task.params)).to.eql({
+        adHocRunParamsId: backfillId,
+        spaceId,
+      });
+
+      // get the execute-backfill events
+      const events: IValidatedEvent[] = await waitForEventLogDocs(
+        backfillId,
+        spaceId,
+        new Map([['execute-backfill', { equal: 4 }]])
```

Comment:
optimistically, each task poll could run one of these, right?  So, would be at least 16 seconds.  Not terrible, but wanted to make sure it wasn't going to take much more than that ...

Reply:
Yes, optimistically, each task poll could run one of these! I could reduce it if we feel it is too long...

Reply:
np; we probably have ones that are even longer :-) - some day we should take a look at how long all these tests take - I never seem to run them locally anymore ...